### PR TITLE
3 - seed rhythm data on deploy/rebase - Man Sang Sin

### LINKED ISSUE #3: Seed rhythm data on deploy/rebase
## Technical task

This ensures that the project is always deployed fresh, with up to date DDL and only seeded data.

Additionally, when a commit changes the DDL, locally this gets reset

## Acceptance Criteria

1. Given _
   - When the project is deployed
   - Then a database is created/reset
   - And it is seeded with the data (DDL and DML) above
2. Given a seeding file exists
   - And I, as a developer, rebase onto main
   - When `npm [fill in here]` is executed
   - Then my database is created/reset
   - And it is seeded with the data from the seeding file

## Implementation Notes

1.
2.

## Checklist:

- [ ] I have defined the user story
- [ ] I have defined some acceptance criteria
- [ ] I have included any implementation notes
- [ ] I have included all relevant links

### PR DESCRIPTION
**Pull request by ManSangSin**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

- modified docker file to include a seed file on first initialisation
- created a seed file
- defined new npm scripts to package.json to start/stop docker
- added a readme with instructions to start/stop docker and info about seed file 

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #3 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

**Comment by mferryRV** [comment]
Hey @ManSangSin - in the future please do not merge PRs without approval. Any team with enough devs will use branch protection and a "four eyes process" meaning you and at least one other person need to look at code before it goes into production.

File: Local Deployment Instructions.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Default database structure:
+  - table name: rhythms
+  - table column names:
+    - id (serial - primary key)
+    - title (varchar - with character limit of 255 imposed)
+    - url (varchar - with character limit of 512 imposed)
+    - location (varchar - with character limit of 255 imposed)
+
+How to start project:
```
Comment (mferryRV) [inline Local Deployment Instructions.md:21]: Good instructions! If you want these displayed in the repo, you will need to add me to the project-level README.md

File: build-rhythms.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+CREATE TABLE rhythms (
+    id		    serial PRIMARY KEY,
+    title	    VARCHAR(255),
+    url		    VARCHAR(512),
+    location	VARCHAR(255)
+);
+
+INSERT INTO rhythms (title, url, location) VALUES('DC Hand dancing','https://www.youtube.com/watch?v=M6uM0qrjetQ','washington DC');
+INSERT INTO rhythms (title, url, location) VALUES('Rumba Guaguanco','https://www.youtube.com/watch?v=gJVT_5swkhA','Havana, Cuba');
```
Comment (mferryRV) [inline build-rhythms.sql:14]: Is there any way to insert multiple values in the same `INSERT` statement?

File: package.json [hunk 3]
```diff
@@ -20,7 +20,12 @@
 		"start": "node dist/server.js",
 		"lint:eslint": "eslint .",
 		"lint:fix": "npm run lint:eslint -- --fix && npm run lint:prettier -- --write",
-		"lint:prettier": "prettier ."
+		"lint:prettier": "prettier .",
+		"docker:run": "docker compose up --detach",
+		"docker:stop": "docker-compose down --volumes",
+		"devtest": "npm run docker:stop && npm run docker:run && npm run dev"
```
Comment (mferryRV) [inline package.json:26]: Do you want the team to move to using `devtest` instead of `dev`? Or is this an additional process?

**Review by mferryRV — COMMENTED** [review]
Overall this does the job for local docker usage but I have made some notes that I would recommend implementing.

Also, please grant me maintainer rights to the repo or implement branch protection on `main`.

**Comment by ManSangSin** [comment]
> Hey @ManSangSin - in the future please do not merge PRs without approval. Any team with enough devs will use branch protection and a "four eyes process" meaning you and at least one other person need to look at code before it goes into production.

Sorry, I wasn't sure how we were suppose to handle the PRs. I should've double checked before merging my code. It won't happen again

**Comment by ManSangSin** [comment]
> Overall this does the job for local docker usage but I have made some notes that I would recommend implementing.
> 
> Also, please grant me maintainer rights to the repo or implement branch protection on `main`.

I've enabled branch protection on "main" but I can't seem to assign maintainer rights, it looks like I need to create an organisation first

File: build-rhythms.sql (hunk 2 above)
Comment (ManSangSin) [reply build-rhythms.sql:14]: We can just have all the values in a list separated by a comma after the VALUES keyword

File: package.json (hunk 3 above)
Comment (ManSangSin) [reply package.json:26]: Seems like an additional process. Its probably better to use/edit the original dev script instead as its the command every is used to for running the project

**Pull request merged**